bluebottle/funding_stripe/views.py

```diff
@@ -90,6 +93,123 @@ class StripePaymentList(PaymentList):
+        stripe = get_stripe()
+        connect_account = donation.activity.bank_account.connect_account
+        payment_method_options = {}
+
+        # Set up payment method options based on currency
+        if currency == 'EUR':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "eu_bank_transfer",
+                        "eu_bank_transfer": {"country": "NL"},
+                    },
+                },
+            }
+        elif currency == 'USD':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "us_bank_transfer",
+                    },
+                },
+            }
+        elif currency == 'GBP':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "gb_bank_transfer",
+                    },
+                },
+            }
+        elif currency == 'MXN':
+            payment_method_options = {
+                "customer_balance": {
+                    "funding_type": "bank_transfer",
+                    "bank_transfer": {
+                        "type": "mx_bank_transfer",
+                    },
+                },
+            }
+
+        # Create the customer in Stripe
+        user = get_current_user()
+        customer = stripe.Customer.create(
+            stripe_account=connect_account.account_id,
+            name=user.full_name,
+            email=user.email,
+        )
+
+        # Create the payment method in Stripe
+        payment_method = stripe.PaymentMethod.create(
+            type="customer_balance",
+            stripe_account=connect_account.account_id,
+        )
+
+        # Define statement descriptor
+        statement_descriptor = connection.tenant.name[:22]
+
+        # Create the PaymentIntent in Stripe
+        intent = stripe.PaymentIntent.create(
+            amount=int(donation.amount.amount * 100),
+            currency=currency,
+            statement_descriptor=statement_descriptor,
+            statement_descriptor_suffix=statement_descriptor[:18],
+            metadata={
+                "tenant_name": connection.tenant.client_name,
+                "tenant_domain": connection.tenant.domain_url,
+                "activity_id": donation.activity.pk,
+                "activity_title": donation.activity.title,
+            },
+            stripe_account=connect_account.account_id,
+            payment_method=payment_method.id,
+            payment_method_options=payment_method_options,
+            payment_method_types=["customer_balance"],
+            customer=customer.id,
+            confirm=True
```

Comment:
A lot of this is also duplicate from the model, can't we merge those code paths somehow?